error-prone-contrib/src/test/java/tech/picnic/errorprone/bugpatterns/JUnitFactoryMethodDeclarationTest.java

```diff
@@ -0,0 +1,280 @@
+            "",
+            "  private static Stream<Arguments> method4TestCases() {",
+            "    /* { foo, bar, baz } */",
+            "    return Stream.of(arguments(1, true, \"A\"), arguments(2, false, \"B\"));",
+            "  }",
+            "",
+            "  @ParameterizedTest",
+            "  @MethodSource(\"testCasesForMethod5\")",
+            "  void method5(int foo, boolean bar, String baz) {}",
+            "",
+            "  void method5TestCases() {}",
+            "",
+            "  // BUG: Diagnostic contains: (but note that a method named `method5TestCases` already exists in",
+            "  // this class)",
+            "  private static Stream<Arguments> testCasesForMethod5() {",
+            "    /* { foo, bar, baz } */",
+            "    return Stream.of(arguments(1, true, \"A\"), arguments(2, false, \"B\"));",
+            "  }",
+            "",
+            "  @ParameterizedTest",
+            "  @MethodSource(\"method6TestCases\")",
+            "  void method6(int foo, boolean bar, String baz) {}",
+            "",
+            "  private static Stream<Arguments> method6TestCases() {",
+            "    List<Arguments> arguments = List.of(arguments(1, true, \"A\"), arguments(2, false, \"B\"));",
+            "    /* { foo, bar, baz } */",
+            "    return arguments.stream();",
+            "  }",
+            "",
+            "  @ParameterizedTest",
+            "  @MethodSource(\"method7TestCases\")",
+            "  void method7(int foo, boolean bar, String baz) {}",
+            "",
+            "  private static Stream<Arguments> method7TestCases() {",
+            "    List<Arguments> arguments = List.of(arguments(1, true, \"A\"), arguments(2, false, \"B\"));",
+            "    // BUG: Diagnostic contains: The return statement should be prefixed by a comment giving the",
+            "    // names of the test case parameters",
+            "    return arguments.stream();",
+            "  }",
+            "",
+            "  private static Stream<Arguments> method8TestCases() {",
+            "    /* { foo, bar, baz } */",
+            "    return Stream.of(arguments(1, true, \"A\"), arguments(2, false, \"B\"));",
+            "  }",
+            "",
+            "  @ParameterizedTest",
+            "  @MethodSource(\"method8TestCases\")",
+            "  void method8(int foo, boolean bar, String baz) {}",
+            "}")
+        .doTest();
+  }
+
+  @Test
+  void replacement() {
+    refactoringTestHelper
+        .addInputLines(
+            "A.java",
+            "import static org.junit.jupiter.params.provider.Arguments.arguments;",
+            "",
+            "import java.util.List;",
+            "import java.util.stream.Stream;",
+            "import org.junit.jupiter.params.ParameterizedTest;",
+            "import org.junit.jupiter.params.provider.Arguments;",
+            "import org.junit.jupiter.params.provider.MethodSource;",
+            "",
+            "class A {",
```

Comment:
In the `identification` test we already nicely show what exact cases are flagged. What we usually do in the `replacement` test is verify the actual rewrite for one or two cases. In this example I _think_ we can get away with using only one example in the `replacement` test that shows all the rewrites. This means we can simplify the test code a lot.